### PR TITLE
update chart feature gates of components

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
The featureGates  of  deploy files `airfacts/deploy` is out of sync with the chart template `charts/karmada/templates`.

Exec :

```shell
helm template charts/karmada | grep feature-gates  -C 3
```

the test results:
```shell
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/chenwen/.kube/config
            - --audit-log-path=-
            - --audit-log-maxage=0
            - --audit-log-maxbackup=0
            - --feature-gates=APIPriorityAndFairness=false,CustomizedClusterResourceModeling=true
          resources:
            requests:
              cpu: 100m
--
            - --bind-address=0.0.0.0
            - --secure-port=10351
            - --leader-elect-resource-namespace=default
            - --feature-gates=CustomizedClusterResourceModeling=true
          livenessProbe:
            httpGet:
              path: /healthz
--
            - --secure-port=10357
            - --leader-elect-resource-namespace=default
            - --v=2
            - --feature-gates=CustomizedClusterResourceModeling=true,Failover=true,GracefulEviction=true,PropagateDeps=ture
          livenessProbe:
            httpGet:
              path: /healthz
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
please correct me if not.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@Poor12 PTAL

File: charts/karmada/templates/_helpers.tpl [hunk 1]
```diff
@@ -390,10 +390,15 @@ Return the proper karmada kubectl image name
 {{ include "common.images.image" (dict "imageRoot" .Values.kubectl.image "global" .Values.global) }}
 {{- end -}}
 
-{{- define "karmada.controllerManager.featureGates" -}}
-     {{- if (not (empty .Values.controllerManager.featureGates)) }}
+{{/* vim: set filetype=mustache: */}}
```
Comment (jwcesign) [inline charts/karmada/templates/_helpers.tpl:393]: Is this necessary? or redundant?

Comment (calvin0327) [reply charts/karmada/templates/_helpers.tpl]: Is't redundant content.

had removed it.

**Comment by jwcesign** [comment]
Nice work!

**Comment by Poor12** [comment]
For the feature gate related to Karmada, I think the default should be empty because 1. If you need to set it explicitly, it means this feature is still in unstable stage. I think does not have to be exposed in charts. 2. This also leads to frequent revisions.

**Pull request closed without merging**